assets/js/modules/pagespeed-insights/components/DashboardPageSpeed.test.js

```diff
@@ -0,0 +1,111 @@
+
+/**
+ * Internal dependencies
+ */
+import DashboardPageSpeed from './DashboardPageSpeed';
+import { fireEvent, render } from '../../../../../tests/js/test-utils';
+import { STORE_NAME, STRATEGY_MOBILE, STRATEGY_DESKTOP } from '../datastore/constants';
+import { STORE_NAME as CORE_SITE } from '../../../googlesitekit/datastore/site/constants';
+import * as fixtures from '../datastore/__fixtures__';
+import fetchMock from 'fetch-mock';
+
+// TODO: update the active class.
+const activeClass = 'googlesitekit-cta-link--danger';
+const url = fixtures.pagespeedMobile.loadingExperience.id;
+const setupRegistry = ( { dispatch } ) => {
+	dispatch( STORE_NAME ).receiveGetReport( fixtures.pagespeedMobile, { url, strategy: STRATEGY_MOBILE } );
+	dispatch( STORE_NAME ).receiveGetReport( fixtures.pagespeedDesktop, { url, strategy: STRATEGY_DESKTOP } );
+	dispatch( CORE_SITE ).receiveSiteInfo( { referenceSiteURL: url } );
+};
+const setupNoReports = ( { dispatch } ) => {
+	dispatch( CORE_SITE ).receiveSiteInfo( { referenceSiteURL: url } );
+};
+const setupRegistryNoFieldDataDesktop = ( { dispatch } ) => {
+	// eslint-disable-next-line no-unused-vars
+	const { metrics, ...desktopLoadingExperience } = fixtures.pagespeedDesktop.loadingExperience;
+	dispatch( STORE_NAME ).receiveGetReport( fixtures.pagespeedMobile, { url, strategy: STRATEGY_MOBILE } );
+	dispatch( STORE_NAME ).receiveGetReport( {
+		...fixtures.pagespeedDesktop,
+		loadingExperience: desktopLoadingExperience, // no field data metrics
+	}, { url, strategy: STRATEGY_DESKTOP } );
+	dispatch( CORE_SITE ).receiveSiteInfo( { referenceSiteURL: url } );
+};
+
+describe( 'DashboardPageSpeed', () => {
+	afterEach( fetchMock.mockClear );
+
+	it( 'renders a progress bar while reports are requested', () => {
+		fetchMock.get(
+			/^\/google-site-kit\/v1\/modules\/pagespeed-insights\/data\/pagespeed/,
+			new Promise( () => {} ), // Don't return a response.
+		);
+		const { queryByRole } = render( <DashboardPageSpeed />, { setupRegistry: setupNoReports } );
+
+		expect( queryByRole( 'progressbar' ) ).toBeInTheDocument();
+	} );
+
+	it( 'displays field data by default when available in both mobile and desktop reports', () => {
+		expect( fixtures.pagespeedMobile.loadingExperience ).toHaveProperty( 'metrics' );
+		expect( fixtures.pagespeedDesktop.loadingExperience ).toHaveProperty( 'metrics' );
```

Comment:
May be unnecessary to assert something on our own fixtures here?

Reply:
It's not intended to test the fixture but to provide context for the test. It saves you from looking into the fixture for context and would be a more clear failure if the fixture were updated in the future with data that didn't have it.